streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -611,65 +624,164 @@ public StateStore getStateStore(final String name) {
     }
 
     /**
-     * Get the {@link KeyValueStore} with the given name.
+     * Get the {@link KeyValueStore} or {@link TimestampedKeyValueStore} with the given name.
      * The store can be a "regular" or global store.
      * <p>
+     * If the registered store is a {@link TimestampedKeyValueStore} this method will return a value-only query
```

Comment:
Meta comment: I'm not sure if it is necessary to maintain compatibility in the test-utils code.

If you have a strong opinion for keeping this, then at least let's add one more comment here that users are recommended to make their test code changes sooner since otherwise it may leads to bad test coverage (e.g. your code actually has a bug that would result in incorrect timestamps, but because of you'd never need to make code changes here you may never capture it).

Reply:
I don't see a difference in `test-utils` package compare to other code -- it's also public API and we should be compatible -- that was the overall purpose of adding a `test-utils` package.

Will extend the comment and a WARN log statement that the test code should be updated.

Reply:
:+1: on preserving compatibility. Also consider that the tests are more useful if the test-driver is roughly analogous to Streams.